frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -1611,6 +1613,93 @@ function ReaderHighlight:getHighlightBookmarkItem()
     end
 end
 
+function ReaderHighlight:_getRollingHighlightInsertionIndex(page, highlight)
+    local position
+    local highlight_beginning = highlight.pos0
+    local highlight_end = highlight.pos1
+    local page_highlights = self.view.highlight.saved[page]
+    for i, cur_highlight in ipairs(page_highlights) do
+        local order = self.ui.document:compareXPointers(highlight_beginning, cur_highlight.pos0)
+        if order == 0 then
+            order = self.ui.document:compareXPointers(highlight_end, cur_highlight.pos1)
+        end
+        if order == nil then
+            break
+        end
+        if order > 0 then
+            position = i
+            break
+        end
+    end
+    if position == nil then
+        position = #page_highlights + 1
+    end
+    return position
+end
+
+function ReaderHighlight:_sortHighlightBounds(page, highlight)
+    local highlight_beginning = highlight.pos0
+    local highlight_end = highlight.pos1
+    highlight_beginning.page = page
+    highlight_end.page = page
+    local order = self.ui.document:comparePositions(highlight_beginning, highlight_end)
+    if order > 0 then
+        return highlight_beginning, highlight_end
+    else
+        return highlight_end, highlight_beginning
+    end
+end
+
+function ReaderHighlight:_getPagingHighlightInsertionIndex(page, highlight)
+    local position
+    local page_highlights = self.view.highlight.saved[page]
+    local highlight_beginning, highlight_end = self:_sortHighlightBounds(page, highlight)
+    for i, cur_highlight in ipairs(page_highlights) do
+        local cur_pos, cur_pos_end = self:_sortHighlightBounds(page, cur_highlight)
+        local order = self.ui.document:comparePositions(highlight_beginning, cur_pos)
+        if order == 0 then
+            order = self.ui.document:comparePositions(highlight_end, cur_pos_end)
+        end
+        if order == nil then
```

Comment:
Not needed, cannot be nil:
https://github.com/koreader/koreader/blob/8c58541c59bac3beb78f4dbe8a7dcf27ab732e16/frontend/document/koptinterface.lua#L1242-L1260

Reply:
Cool! Removed the nil check in this place.